common/src/main/java/org/apache/sedona/common/Predicates.java

```diff
@@ -62,4 +64,82 @@ public static boolean dWithin(Geometry leftGeometry, Geometry rightGeometry, dou
             return leftGeometry.isWithinDistance(rightGeometry, distance);
         }
     }
+
+    /**
+     * Checks if a geometry crosses the International Date Line.
+     *
+     * @param geometry The geometry to check.
+     * @return True if the geometry crosses the Date Line, false otherwise.
+     */
+    public static boolean crossesDateLine(Geometry geometry) {
+        if (geometry == null || geometry.isEmpty()) {
+            return false;
+        }
+
+        AtomicBoolean crossesDateLine = new AtomicBoolean(false);
```

Comment:
Is there a reason why use `AtomicBoolean`? I believe the filter functions will be called in a sequential order?

Reply:
The `CoordinateSequenceFilter` being an anonymous inner class could not modify a local variable outside its filter scope unless its `final`. `AtomicBoolean` was a work around because final variable's cannot be modified. I have removed separate handling of `GeometryCollection` so I can define a private local variable `crossesDateLine` within the filter.